cmd/timescaledb-parallel-copy/main.go

```diff
@@ -86,29 +87,38 @@ func (l csvCopierLogger) Infof(msg string, args ...interface{}) {
 
 func main() {
 	if showVersion {
-		fmt.Printf("%s %s (%s %s)\n", binName, version, runtime.GOOS, runtime.GOARCH)
+		log.Printf("%s %s (%s %s)\n", binName, version, runtime.GOOS, runtime.GOARCH)
 		os.Exit(0)
 	}
 
+	if dbName != "" {
+		log.Fatalf("Error: Deprecated flag -db-name is being used. Update -connection to connect to the given database")
+	}
+	opts := []csvcopy.Option{
+		csvcopy.WithLogger(&csvCopierLogger{}),
+		csvcopy.WithSchemaName(schemaName),
+		csvcopy.WithCopyOptions(copyOptions),
+		csvcopy.WithSplitCharacter(splitCharacter),
+		csvcopy.WithQuoteCharacter(quoteCharacter),
+		csvcopy.WithEscapeCharacter(escapeCharacter),
+		csvcopy.WithColumns(columns),
+		csvcopy.WithWorkers(workers),
+		csvcopy.WithLimit(limit),
+		csvcopy.WithBatchSize(batchSize),
+		csvcopy.WithLogBatches(logBatches),
+		csvcopy.WithReportingPeriod(reportingPeriod),
+		csvcopy.WithVerbose(verbose),
+	}
+	if headerLinesCnt > 1 {
```

Comment:
Shouldn't this be:
```suggestion
	if headerLinesCnt > 0 {
```

Otherwise you'll enter the fail branch of `WithSkipHeaderCount`:

```go
if headerLineCount == 0 {
			return errors.New("header line count must be greater than zero")
}
```

Another thing, this changes the current behavior. I'm not saying the pass behavior is the best either. The old behavior is, if `skip == false` then it doesn't matter the value of `headerLinesCnt`, nothing is going to be skipped. By default headers are not skipped.

With you changes, since `headerLinesCnt` defaults to `1`, and you are not requiring `skip == true` to set the value for skip, then we are defaulting to always skip the first line.

I'd rather not change the old behavior, and if we do it, let's document it.